src/components/Comment.jsx

```diff
@@ -0,0 +1,187 @@
+        if (res.ok) {
+          return res.json();
+        }
+        throw new Error(`Something went wrong! Status: ${res.status}`);
+      })
+      .then(() => {
+        postContext.setComments([...updatedList]);
+      })
+      .catch((err) => {
+        console.log("error occured: ", err);
+      });
+
+    setFormData(formData);
+    setUpdate(false);
+  };
+
+  useState(() => {
+    const thisUser = postContext.contacts.find(
+      (x) => parseInt(x.id) === parseInt(comment.contactId)
+    );
+    setUser(thisUser);
+  }, [comment]);
+
+  useEffect(() => {
+    const thisUser = postContext.contacts.find(
+      (x) => parseInt(x.id) === parseInt(comment.contactId)
+    );
+    setUser(thisUser);
+  }, [postContext.contacts, comment, postContext.comments]);
+
+  return (
+    <div className="comment-container">
+      {user === undefined || user === "" ? (
+        <p>loading...</p>
+      ) : (
+        <Icon user={user} />
+      )}
+      <div className="div-comment">
+        {user === undefined || user === "" ? (
+          <p>loading...</p>
+        ) : (
+          <div className="comment-background">
+            <button
+              className="del-button"
+              onClick={(e) => {
+                e.preventDefault();
+                handleDeleteComment(comment.id);
+              }}
+            >
+              Delete
+            </button>
+            <button
+              className="modify-btn"
+              onClick={(e) => {
+                e.preventDefault();
+                setFormData(comment);
+                setUpdate(!update);
+              }}
+            >
+              Modify
+            </button>
+            <Link to={`/profile/${user.id}`} className="profile-link">
+              <h4>{user.firstName + " " + user.lastName} </h4>
+            </Link>
+            {update ? (
+              <form
+                className="update-comment-form"
+                onSubmit={(e) => {
+                  e.preventDefault();
+                  handleUpdateComment(formData);
+                }}
+              >
+                <div className="update-comment-field">
+                  <input
+                    className="update-comment-input"
+                    id="content"
+                    name="content"
+                    type="text"
+                    placeholder={`${formData.content}`}
+                    value={formData.content ?? ""}
+                    onChange={handleInputChange}
+                  ></input>
+                  <button type="submit" className="img-update-btn">
+                    <img className="update-button-img" src={sendIcon} />
+                  </button>
+                </div>
+              </form>
+            ) : (
+              <p>{comment.content}</p>
+            )}
+          </div>
+        )}
+      </div>
+    </div>
+  );
+}
+
+export default Comment;
+
+Comment.propTypes = {
+  comment: PropTypes.object,
+  comments: PropTypes.array,
```

Comment:
These 3 props do not appear to be used within the component. It seems you're using context to access similar data. Remove any unused props to clean up your component.